shared/components/button/Button.tsx

```diff
@@ -1,12 +1,18 @@
-import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-const Button = styled.button`
-  background-color: ${props =>
+type ButtonProps = {
+  type: string;
```

Comment:
You could also do:
`type?: 'submit'`

Then it's optional and by default you'll style it as `regular` preventing you from having to add a `type` always and also it's more strict.